code/modules/keybindings/bindings_atom.dm

```diff
@@ -0,0 +1,45 @@
+// You might be wondering why this isn't client level. If focus is null, we don't want you to move.
+// Only way to do that is to tie the behavior into the focus's keyLoop().
+
+// THE TRADITIONAL STYLE FROM /TG (modified)
+
+/atom/movable/keyLoop(client/user)
+	// Bail out if the user is holding the "face direction" key (Maybe?)
+	// TODO - I think this breaks non-hotkeys WASD movement, so maybe adopt the /tg solution)
+	if(user.mod_keys_held & CTRL_KEY)
+		return
+
+	var/must_call_move = FALSE
+	var/movement_dir = MOVEMENT_KEYS_TO_DIR(user.move_keys_held)
+	if(user.next_move_dir_add)
+		must_call_move = TRUE // So that next_move_dir_add gets cleared if its time.
+		movement_dir |= user.next_move_dir_add
+	if(user.next_move_dir_sub)
+		DEBUG_INPUT("[(user.next_move_dir_sub & movement_dir) ? "Actually" : "Want to"] subtract [dirs2text(user.next_move_dir_sub)] from [dirs2text(movement_dir)]")
```

Comment:
Where does this get output to? How often will this generate logs?

Reply:
```
// Uncomment to get a lot of debug logging for movement keys.
// #define DEBUG_INPUT(A) to_world_log(A)
#define DEBUG_INPUT(A) 
```
From elsewhere in the PR.

I'm not convinced this is a good pattern to include, and would be better handled at least with an ifdef rather telling the programmer to switch line commenting, but:

When in the uncommented form, is modified to an empty line by the compiler (and thus creates no work and has no outputs). When in the commented form, outputs directly to the world log.